src/library/commissioner_impl.cpp

```diff
@@ -658,8 +657,16 @@ void CommissionerImpl::CommandDiagGetQuery(ErrorHandler aHandler, const std::str
     }
     else
     {
+        Address dstAddr;
         SuccessOrExit(error = dstAddr.Set(aAddr));
-        mProxyClient.SendRequest(request, onResponse, dstAddr, kDefaultMmPort);
```

Comment:
Should other commissioner.hpp APIs also support RLOC16 addresses? Is only this `CommandDiagGetQuery()` API which can support RLOC16?